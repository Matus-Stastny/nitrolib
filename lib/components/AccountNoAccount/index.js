"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _Heading = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Heading"));

var _ButtonLink = _interopRequireDefault(require("@kiwicom/orbit-components/lib/ButtonLink"));

var _Button = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Button"));

var _Illustration = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Illustration"));

var _Facebook = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/Facebook"));

var _Google = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/Google"));

var _ChevronLeft = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/ChevronLeft"));

var _ModalHeader = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Modal/ModalHeader"));

var _ModalSection = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Modal/ModalSection"));

var _Stack = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Stack"));

var _Translate = _interopRequireDefault(require("../Translate"));

var _Text = _interopRequireDefault(require("../Text"));

var _context = require("../../services/brand/context");

var _ref2 =
/*#__PURE__*/
React.createElement(_Illustration.default, {
  name: "NoResults",
  size: "small"
});

var _ref3 =
/*#__PURE__*/
React.createElement(_Heading.default, {
  element: "h2"
}, React.createElement(_Translate.default, {
  t: "account.no_bookings_or_account"
}));

var _ref5 =
/*#__PURE__*/
React.createElement(_Text.default, {
  t: "account.or_social_account",
  weight: "bold",
  spaceAfter: "medium"
});

var _ref6 =
/*#__PURE__*/
React.createElement(_Facebook.default, null);

var _ref7 =
/*#__PURE__*/
React.createElement(_Google.default, null);

var _ref8 =
/*#__PURE__*/
React.createElement(_Translate.default, {
  t: "account.register"
});

var _ref9 =
/*#__PURE__*/
React.createElement(_ChevronLeft.default, null);

var _ref10 =
/*#__PURE__*/
React.createElement(_Translate.default, {
  t: "account.back"
});

var AccountNoAccount = function AccountNoAccount(_ref) {
  var onBack = _ref.onBack,
      onRegister = _ref.onRegister,
      onFacebookLogin = _ref.onFacebookLogin,
      onGoogleLogin = _ref.onGoogleLogin;

  var _ref4 =
  /*#__PURE__*/
  React.createElement(_ModalSection.default, null, React.createElement(_Stack.default, {
    spacing: "natural",
    direction: "column"
  }, React.createElement(_Button.default, {
    onClick: onRegister
  }, _ref8), React.createElement(_ButtonLink.default, {
    type: "secondary",
    iconLeft: _ref9,
    onClick: onBack
  }, _ref10)));

  return React.createElement(_context.Consumer, null, function (brand) {
    return React.createElement(React.Fragment, null, React.createElement(_ModalHeader.default, null, _ref2, _ref3, React.createElement(_Text.default, {
      t: "account.no_bookings_or_account_description",
      values: {
        brandName: brand.name
      }
    })), _ref4, React.createElement(_ModalSection.default, {
      suppressed: true
    }, _ref5, React.createElement(_Stack.default, {
      spacing: "natural",
      flex: true
    }, React.createElement(_Button.default, {
      type: "facebook",
      block: true,
      bordered: true,
      icon: _ref6,
      onClick: onFacebookLogin
    }, React.createElement(_Translate.default, {
      t: "account.log_in_with",
      values: {
        provider: "Facebook"
      }
    })), React.createElement(_Button.default, {
      type: "google",
      block: true,
      bordered: true,
      icon: _ref7,
      onClick: onGoogleLogin
    }, React.createElement(_Translate.default, {
      t: "account.log_in_with",
      values: {
        provider: "Google"
      }
    })))));
  });
};

var _default = AccountNoAccount;
exports.default = _default;