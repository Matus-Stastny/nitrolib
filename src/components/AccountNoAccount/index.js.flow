// @flow strict
import * as React from "react";

type Props = {|
  brandName: string,
  onBack: (ev: SyntheticEvent<HTMLDivElement> | SyntheticEvent<HTMLButtonElement>) => void,
  onRegister: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onFacebookLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onGoogleLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
|};

declare export default React.ComponentType<Props>;
