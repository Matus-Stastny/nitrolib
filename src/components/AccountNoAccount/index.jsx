// @flow strict
import * as React from "react";
import styled from "styled-components";
import OrbitText from "@kiwicom/orbit-components/lib/Text";
import Heading from "@kiwicom/orbit-components/lib/Heading";
import ButtonLink from "@kiwicom/orbit-components/lib/ButtonLink";
import Button from "@kiwicom/orbit-components/lib/Button";
import Illustration from "@kiwicom/orbit-components/lib/Illustration";
import FacebookIcon from "@kiwicom/orbit-components/lib/icons/Facebook";
import GoogleIcon from "@kiwicom/orbit-components/lib/icons/Google";
import ChevronLeft from "@kiwicom/orbit-components/lib/icons/ChevronLeft";

import Flex from "../../primitives/Flex";
import Text from "../Text";
import type { ThemeProps } from "../../records/Theme";

const SpacingXSmall = styled.div`
  margin-bottom: 8px;
`;

const SpacingMedium = styled.div`
  margin-bottom: 16px;
`;

const SpacingBig = styled.div`
  margin-bottom: 32px;
`;

const Rectangle = styled.div`
  border-top: 1px solid ${({ theme }: ThemeProps) => theme.orbit.backgroundSeparator};
  width: 100%;
  height: 128px;
  background-color: ${({ theme }: ThemeProps) => theme.orbit.backgroundBody};
  padding-top: 24px;
  padding-bottom: 24px;
  margin-top: 50px;
  margin-bottom: 16px;
`;

type Props = {|
  onBack: (ev: SyntheticEvent<HTMLDivElement> | SyntheticEvent<HTMLButtonElement>) => void,
  onRegister: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onFacebookLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onGoogleLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
|};

const AccountNoAccount = ({ onBack, onRegister, onFacebookLogin, onGoogleLogin }: Props) => (
  <React.Fragment>
    <Illustration name="NoBookings" size="small" />
    <SpacingXSmall>
      <Heading element="h2">
        <Text t={__("account.no_bookings_or_account")} />
      </Heading>
    </SpacingXSmall>
    <SpacingBig>
      <OrbitText>
        <Text t={__("account.no_bookings_or_account_description")} />
      </OrbitText>
    </SpacingBig>
    <SpacingMedium>
      <Button onClick={onRegister}>
        <Text t={__("account.register")} />
      </Button>
    </SpacingMedium>
    <SpacingMedium>
      <ButtonLink type="secondary" iconLeft={<ChevronLeft />} onClick={onBack}>
        <Text t={__("account.back")} />
      </ButtonLink>
    </SpacingMedium>
    <Rectangle>
      <OrbitText weight="bold">
        <Text t={__("account.or_social_account")} />
      </OrbitText>
      <Flex>
        <span style={{ flexGrow: 1, marginRight: "4px" }}>
          <Button type="facebook" block bordered icon={<FacebookIcon />} onClick={onFacebookLogin}>
            <Text t={__("account.log_in_with")} values={{ provider: "Facebook" }} />
          </Button>
        </span>
        <span style={{ flexGrow: 1, marginLeft: "4px" }}>
          <Button type="google" block bordered icon={<GoogleIcon />} onClick={onGoogleLogin}>
            <Text t={__("account.log_in_with")} values={{ provider: "Google" }} />
          </Button>
        </span>
      </Flex>
    </Rectangle>
  </React.Fragment>
);

export default AccountNoAccount;
