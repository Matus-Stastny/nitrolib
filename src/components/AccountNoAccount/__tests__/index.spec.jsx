// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import AccountNoAccount from "..";

describe("#AccountNoAccount", () => {
  test("render", () => {
    const wrapper = shallow(
      <AccountNoAccount
        onBack={() => {}}
        onRegister={() => {}}
        onFacebookLogin={() => {}}
        onGoogleLogin={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
