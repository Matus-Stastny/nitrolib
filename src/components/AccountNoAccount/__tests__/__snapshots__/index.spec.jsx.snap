// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AccountNoAccount render 1`] = `
<Fragment>
  <ModalHeader>
    <Illustration
      name="NoBookings"
      size="small"
    />
    <Heading
      element="h2"
    >
      <Text
        html={false}
        t="account.no_bookings_or_account"
        values={Object {}}
      />
    </Heading>
    <Text>
      <Text
        html={false}
        t="account.no_bookings_or_account_description"
        values={Object {}}
      />
    </Text>
  </ModalHeader>
  <ModalSection>
    <Stack
      desktop={
        Object {
          "spacing": "natural",
        }
      }
    >
      <Button
        onClick={[Function]}
      >
        <Text
          html={false}
          t="account.register"
          values={Object {}}
        />
      </Button>
      <AccountNoAccount__Align>
        <ButtonLink
          component="button"
          external={false}
          iconLeft={<ChevronLeft />}
          onClick={[Function]}
          size="normal"
          transparent={false}
          type="secondary"
          width={0}
        >
          <Text
            html={false}
            t="account.back"
            values={Object {}}
          />
        </ButtonLink>
      </AccountNoAccount__Align>
    </Stack>
  </ModalSection>
  <ModalSection
    suppressed={true}
  >
    <Text
      spaceAfter="medium"
      weight="bold"
    >
      <Text
        html={false}
        t="account.or_social_account"
        values={Object {}}
      />
    </Text>
    <Stack
      desktop={
        Object {
          "direction": "row",
          "spacing": "natural",
        }
      }
    >
      <Button
        block={true}
        bordered={true}
        icon={<Facebook />}
        onClick={[Function]}
        type="facebook"
      >
        <Text
          html={false}
          t="account.log_in_with"
          values={
            Object {
              "provider": "Facebook",
            }
          }
        />
      </Button>
      <Button
        block={true}
        bordered={true}
        icon={<Google />}
        onClick={[Function]}
        type="google"
      >
        <Text
          html={false}
          t="account.log_in_with"
          values={
            Object {
              "provider": "Google",
            }
          }
        />
      </Button>
    </Stack>
  </ModalSection>
</Fragment>
`;
